README.md

```diff
@@ -1,7 +1,26 @@
 # predicators
 
+## Repository Description
+
+This codebase provides a concrete implementation of [Neuro-Symbolic Relational Transition Models](https://arxiv.org/abs/2105.14074) for task and motion planning. **Several features are concurrently under active development -- please contact <tslvr@mit.edu> and <ronuchit@mit.edu> before attempting to use it for your own research.**
+
+The scope of this codebase extends far beyond the scope of the paper linked above. In particular, this codebase aims to ultimately provide an integrated system for learning the ingredients of search-then-sample task and motion planning. That includes: options, predicates, operators, and samplers.
+
+### Code structure
+
+In `src/`, the environments are defined in the `envs/` directory, and the approaches (both learning-based and not) are defined in the `approaches/` directory. The core NSRT learning loop happens in `src/nsrt_learning.py`, which has the following steps:
+* Segment data based on changes in predicates.
+* Learn how many NSRTs we need, along with the symbolic operator components of each (parameters, preconditions, and effects).
+* Learn options and annotate data with them.
+* Learn samplers.
+* Finalize the NSRTs.
+
+Methods for predicate learning are implemented as approaches (e.g., `src/approaches/iterative_invention_approach.py`), and may interface with the core structure of `src/nsrt_learning.py` in various ways.
```

Comment:
Maybe also mention that sampler learning and option learning are just functions, by contrast